### PR TITLE
RFC: Remove Windows CE support

### PR DESCRIPTION
**Pull request by teksturi**
As Windows CE does not support C++17 nor Visual Studio 2017 anymore support for it should be removed. There is still Windows CE documentation which tells situation.

---

RFC for now as I'm not totally sure if you guys have plans to continue support it with somehow. It is totally ok for reject. Note for reviewers. It is easier to review commits as they have logically groped.

### CODE REVIEW

File: doc/99200-WinCEPlatformNotes.page [hunk 1]
```diff
@@ -3,130 +3,7 @@ AAAIntroduction
...
-Specifically, setting <*Optimization*> to <*Maximize Speed (/O2)*> will result in failures in the
-test suite for Foundation Events due to a compiler optimizer bug.
-
-For shared/DLL builds, the /FORCE:MULTIPLE option is passed to the linker. This is
-to avoid a problem with iostream classes and their methods (template instantiations),
-which for some unknown reason (possibly bug) will be exported by the Foundation library
-(and others) and thus cause multiply defined symbol errors.
-
-The reference system used for testing is a Digi ConnectCore 9P 9360 running Windows CE 6.0.
+Starting with release 1.4.0 the POCO C++ Libraries cannot be used on Windows CE
```
Comment (teksturi) [inline doc/99200-WinCEPlatformNotes.page:6]: Need to update version number to something else. Will do this at some point if removing this get's green light and we know if this going to land 1.13.0 or 1.14.0.

**Comment by teksturi** [comment]
I'm also wondering are _vs90* project files files for windows CE?

**Comment by aleks-f** [comment]
> I'm also wondering are _vs90* project files files for windows CE?

No, those are needed for [ProGen](https://github.com/pocoproject/poco/wiki/Generating-VS-Projects-using-ProGen)

**Review by aleks-f — APPROVED** [review]
I'm fine with removing CE. That OS is not even supported by MS anymore

**Review by obiltschnig — APPROVED** [review]
lgtm

**Pull request merged**